src/ITensorChainRules/ITensorChainRules.jl

```diff
@@ -190,22 +190,24 @@ function ChainRulesCore.rrule(::typeof(itensor), x::Array, a...)
   y = itensor(x, a...)
   function itensor_pullback(ȳ)
     x̄ = array(ȳ)
-    ā = broadcast(_ -> NoTangent(), a)
+    ā = broadcast_notangent(a)
     return (NoTangent(), x̄, ā...)
   end
   return y, itensor_pullback
 end
 
-function ChainRulesCore.rrule(::typeof(ITensor), x::Array, a...)
-  y = ITensor(x, a...)
-  function ITensor_pullback(ȳ)
-    # TODO: define `Array(::ITensor)` directly
-    x̄ = Array(ȳ, inds(ȳ)...)
-    ā = broadcast(_ -> NoTangent(), a)
-    return (NoTangent(), x̄, ā...)
-  end
-  return y, ITensor_pullback
-end
+# TODO: this function is triggered in the HVP kernel and triggered error,
+# although I'm still not clear about the reason. Can debug later.
+# function ChainRulesCore.rrule(::typeof(ITensor), x::Array, a...)
```

Comment:
So you see an error when this` rrule` is defined, but it fixes it to remove this definition?

Reply:
yes

Reply:
Reminder to try to add this back with the signature:
```julia
ChainRulesCore.rrule(::typeof(ITensor), x::Array{<:Number}, a...)
```
or more specifically:
```julia
ChainRulesCore.rrule(::typeof(ITensor), x::Array{<:Number}, a::Index...)
```

Reply:
done